homeassistant/components/enphase_envoy/config_flow.py

```diff
@@ -141,9 +141,13 @@ async def async_step_zeroconf(
                 and entry.data[CONF_HOST] == self.ip_address
             ):
                 _LOGGER.debug(
-                    "Zeroconf update envoy with this ip and blank serial in unique_id",
+                    "Zeroconf update envoy with this ip and blank unique_id",
                 )
-                title = f"{ENVOY} {serial}" if entry.title == ENVOY else ENVOY
+                # Found an entry with blank unique_id (prior deleted) with same ip
+                # If the title is still default shorthand 'Envoy' then append serial
+                # to differentiate multiple Envoy. Don't change the title if any other
+                # title is still present in the old entry.
+                title = f"{ENVOY} {serial}" if entry.title == ENVOY else entry.title
                 return self.async_update_reload_and_abort(
                     entry, title=title, unique_id=serial, reason="already_configured"
```

Comment:
I think we can remove the title right?

Reply:
I suspect it's there to make sure if there's multiple envoy's, they will have a unique title. If that would be an issue?

Reply:
I guess that works, we now only conditionally change the title

Reply:
Looks like it was there from some 3 years ago when @bdraco added config flow in #48517 . So it can very well be obsolete these days, but (as we all know) it's not really my expertise.